### ISSUE TITLE
Regression in 0.16.0: solve_banded errors out in patsy test suite

### ISSUE THREAD

**Issue by njsmith** [scipy/scipy#5127]
SciPy 0.15.1:
```
>>> import scipy.linalg
>>> import numpy as np
>>> scipy.linalg.solve_banded((1, 1), np.asarray([[0], [1.0], [0]]), np.asarray([[1.0, 2.0, 3.0]]))
array([[ 1.,  2.,  3.]])
```

SciPy 0.16.0:
```
>>> import scipy.linalg
>>> import numpy as np
>>> scipy.linalg.solve_banded((1, 1), np.asarray([[0], [1.0], [0]]), np.asarray([[1.0, 2.0, 3.0]]))
unexpected array size: new_size=1, got array with arr_size=0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/njs/miniconda/envs/travis-replica-2.6/lib/python2.6/site-packages/scipy/linalg/basic.py", line 234, in solve_banded
    overwrite_ab, overwrite_b)
_flapack.error: failed in converting 1st argument `dl' of _flapack.dgtsv to C/Fortran array
```

This example is derived from a case that occurs in patsy when computing spline bases: https://github.com/pydata/patsy/blob/master/patsy/mgcv_cubic_splines.py#L48

and causes failures in the patsy test suite: https://travis-ci.org/pydata/patsy/builds/74659634

Aside from this appearing to be a regression in scipy, I'd appreciate thoughts on how to work around the bug for users who do have scipy 0.16.0 installed, since I didn't actually write this code and am not actually familiar with `solve_banded` :-)

**Comment by ev-br** [comment]
A sledgehammer of a workaround, if you really need one to work today, could be to copy the solve_banded without the `if l==u== 1:` branch which was added in scipy 0.16. https://github.com/scipy/scipy/blob/master/scipy/linalg/basic.py#L176

That branch special-cases three-diagonal matrices, and one motivation for adding it was precisely cubic splines :-).

**Comment by ewmoore** [comment]
Fix in #5128.
